Userland/Libraries/LibJS/Runtime/RegExpObject.cpp

```diff
@@ -156,48 +156,78 @@ void RegExpObject::initialize(Realm& realm)
 }
 
 // 22.2.3.2.2 RegExpInitialize ( obj, pattern, flags ), https://tc39.es/ecma262/#sec-regexpinitialize
-ThrowCompletionOr<NonnullGCPtr<RegExpObject>> RegExpObject::regexp_initialize(VM& vm, Value pattern, Value flags)
+ThrowCompletionOr<NonnullGCPtr<RegExpObject>> RegExpObject::regexp_initialize(VM& vm, Value pattern_value, Value flags_value)
 {
-    String f;
-    if (flags.is_undefined()) {
-        f = String::empty();
-    } else {
-        f = TRY(flags.to_string(vm));
-    }
-
-    String original_pattern;
-    String parsed_pattern;
-
-    if (pattern.is_undefined()) {
-        original_pattern = String::empty();
-        parsed_pattern = String::empty();
-    } else {
-        original_pattern = TRY(pattern.to_string(vm));
-        bool unicode = f.find('u').has_value();
-        bool unicode_sets = f.find('v').has_value();
-        parsed_pattern = TRY(parse_regex_pattern(vm, original_pattern, unicode, unicode_sets));
-    }
-
-    auto parsed_flags_or_error = regex_flags_from_string(f);
+    // NOTE: This also contains changes adapted from https://arai-a.github.io/ecma262-compare/?pr=2418, which doesn't match the upstream spec anymore.
+
+    // 1. If pattern is undefined, let P be the empty String.
+    // 2. Else, let P be ? ToString(pattern).
+    auto pattern = pattern_value.is_undefined()
+        ? String::empty()
+        : TRY(pattern_value.to_string(vm));
+
+    // 3. If flags is undefined, let F be the empty String.
+    // 4. Else, let F be ? ToString(flags).
+    auto flags = flags_value.is_undefined()
+        ? String::empty()
+        : TRY(flags_value.to_string(vm));
+
+    // 5. If F contains any code unit other than "d", "g", "i", "m", "s", "u", or "y" or if it contains the same code unit more than once, throw a SyntaxError exception.
+    auto parsed_flags_or_error = regex_flags_from_string(flags);
     if (parsed_flags_or_error.is_error())
         return vm.throw_completion<SyntaxError>(parsed_flags_or_error.release_error());
 
+    // 6. If F contains "i", let i be true; else let i be false.
+    // 7. If F contains "m", let m be true; else let m be false.
+    // 8. If F contains "s", let s be true; else let s be false.
+    // 9. If F contains "u", let u be true; else let u be false.
+    // 10. If F contains "v", let v be true; else let v be false.
```

Comment:
Should these steps also be above the `auto parsed_flags_or_error = regex_flags_from_string(flags);` line?

Reply:
Yep, done!